### PR TITLE
[feature]: Able to use save_best option

### PR DESCRIPTION
**Pull request by yinchimaoliang**
Change `eval_hooks.py`, able to use `save_best` while training.

### CODE REVIEW

File: tools/train.py [hunk 1]
```diff
@@ -149,7 +149,6 @@ def main():
             PALETTE=datasets[0].PALETTE)
     # add an attribute for visualization convenience
     model.CLASSES = datasets[0].CLASSES
-    meta.update(cfg.checkpoint_config.meta)
```
Comment (yinchimaoliang) [inline tools/train.py]: This seems to be necessary, without updating meta, the best ckpt can not be used to test.

Comment (xvjiarui) [reply tools/train.py]: Why we need to update meta?

Comment (yinchimaoliang) [reply tools/train.py]: Right now, meta missing some keys such as `CLASSES` and `PALETTE`, so the best ckpt it saves won't have these keys, which will cause failure during test time. Updating meta will solve the problem.

Comment (xvjiarui) [reply tools/train.py]: I see. Reverted.

**Review by xvjiarui — APPROVED** [review]

**Pull request merged**